File: decompiled/3.18.0/be/ehealth/businessconnector/chapterIV/domain/ChapterIVReferences.java
package be.ehealth.businessconnector.chapterIV.domain;

import be.ehealth.technicalconnector.exception.TechnicalConnectorException;
import be.ehealth.technicalconnector.idgenerator.IdGeneratorFactory;
import java.io.Serializable;
import org.apache.commons.lang.builder.EqualsBuilder;

public final class ChapterIVReferences implements Serializable {
   private static final long serialVersionUID = 5945745713020351653L;
   private String commonReference;
   private String commonNIPReference;
   private String kmehrIdSuffix;
   private String recordCommonInputId;

   public ChapterIVReferences(boolean init) throws TechnicalConnectorException {
      if (init) {
         String id = IdGeneratorFactory.getIdGenerator().generateId();
         this.commonReference = id;
         this.recordCommonInputId = id;
         this.kmehrIdSuffix = id;
      }

   }

   public ChapterIVReferences(String id) {
      this.commonReference = id;
      this.recordCommonInputId = id;
      this.kmehrIdSuffix = id;
   }

   public String getCommonReference() {
      return this.commonReference;
   }

   public void setCommonReference(String commonReference) {
      this.commonReference = commonReference;
   }

   public String getCommonNIPReference() {
      return this.commonNIPReference;
   }

   public void setCommonNIPReference(String commonNIPReference) {
      this.commonNIPReference = commonNIPReference;
   }

   public String getKmehrIdSuffix() {
      return this.kmehrIdSuffix;
   }

   public void setKmehrIdSuffix(String kmehrIdSuffix) {
      this.kmehrIdSuffix = kmehrIdSuffix;
   }

   public String getRecordCommonInputId() {
      return this.recordCommonInputId;
   }

   public void setRecordCommonInputId(String recordCommonInputId) {
      this.recordCommonInputId = recordCommonInputId;
   }

   public int hashCode() {
      int prime = true;
      int result = 1;
      int result = 31 * result + (this.commonNIPReference == null ? 0 : this.commonNIPReference.hashCode());
      result = 31 * result + (this.commonReference == null ? 0 : this.commonReference.hashCode());
      result = 31 * result + (this.kmehrIdSuffix == null ? 0 : this.kmehrIdSuffix.hashCode());
      result = 31 * result + (this.recordCommonInputId == null ? 0 : this.recordCommonInputId.hashCode());
      return result;
   }

   public boolean equals(Object obj) {
      if (this == obj) {
         return true;
      } else if (obj == null) {
         return false;
      } else if (this.getClass() != obj.getClass()) {
         return false;
      } else {
         ChapterIVReferences other = (ChapterIVReferences)obj;
         return (new EqualsBuilder()).append(this.commonNIPReference, other.getCommonNIPReference()).append(this.commonReference, other.getCommonReference()).append(this.getKmehrIdSuffix(), other.getKmehrIdSuffix()).append(this.recordCommonInputId, other.getRecordCommonInputId()).isEquals();
      }
   }
}

File: src/main/kotlin/org/taktik/connector/business/chapterIV/domain/ChapterIVReferences.kt
package org.taktik.connector.business.chapterIV.domain

import org.taktik.connector.technical.exception.TechnicalConnectorException
import org.taktik.connector.technical.idgenerator.IdGeneratorFactory
import java.io.Serializable
import org.apache.commons.lang.builder.EqualsBuilder

class ChapterIVReferences @Throws(TechnicalConnectorException::class) constructor(init: Boolean) : Serializable {
    var commonReference: String? = null
    var commonNIPReference: String? = null
    var kmehrIdSuffix: String? = null
    var recordCommonInputId: String? = null

    init {
        if (init) {
            val id = IdGeneratorFactory.getIdGenerator().generateId()
            this.commonReference = id
            this.recordCommonInputId = id
            this.kmehrIdSuffix = id
        }
    }

    override fun hashCode(): Int {
        var result = 1
        result = 31 * result + if (this.commonNIPReference == null) 0 else this.commonNIPReference!!.hashCode()
        result = 31 * result + if (this.commonReference == null) 0 else this.commonReference!!.hashCode()
        result = 31 * result + if (this.kmehrIdSuffix == null) 0 else this.kmehrIdSuffix!!.hashCode()
        result = 31 * result + if (this.recordCommonInputId == null) 0 else this.recordCommonInputId!!.hashCode()
        return result
    }

    override fun equals(other: Any?) = when {
        this === other -> true
        other == null -> false
        this.javaClass != other.javaClass -> false
        else -> {
            val other = other as ChapterIVReferences?
            EqualsBuilder().append(this.commonNIPReference, other!!.commonNIPReference)
                .append(this.commonReference, other.commonReference).append(this.kmehrIdSuffix, other.kmehrIdSuffix)
                .append(this.recordCommonInputId, other.recordCommonInputId).isEquals
        }
    }
}
